### ISSUE TITLE
createCertificate with password-protected private key file

### ISSUE THREAD

**Issue by lordbah** [Dexus/pem#255]
I wanted to create a certificate with a password-protected private key file, and couldn't figure out how to do it. It kept creating an unprotected key file. In the end I had to make a code change to get it to work by adding the options parameter in the call to createPrivateKey on line 209 of pem.js

```
  if (!options.clientKey) {
    createPrivateKey(options.keyBitsize || 2048, options, function (error, keyData) {
```

and had to set the following in the options passed to createCertificate():

```
certOptions.cipher = "aes256";
certOptions.password = "secret";
certOptions.clientKeyPassword = "secret";  // same value as previous line
```

I don't think I understand the code well enough to say this is the right thing to do and won't break anything else. Hoping some other folks can take a look.

**Comment by Dexus** [comment]
I see there is a bug to create a password for the key when you create it with the `pem.createCertificate` function.
Workaround is use `pem.createPrivateKey` together with `options.cipher` and `opions.password` and then use them to create a CSR and then the Certificate.

**Comment by Dexus** [comment]
Now `password` and/or `clientKeyPassword` will supported on direct call of `createCertificate` see 0385fbcdb6e6e8f5dddeb7b32297f80f6bd065cb

Will in the next release included with v1.16.0
